flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -606,25 +606,33 @@ runBashJavaUtilsCmd() {
     local class_path=$2
     local conf_dir=$3
 
-    local output="`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2> /dev/null`"
+    local output=`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2> /dev/null | tail -n 1`
     if [[ $? -ne 0 ]]; then
-        echo "[ERROR] Cannot run BashJavaUtils to execute command ${cmd}."
+        echo "[ERROR] Cannot run BashJavaUtils to execute command ${cmd}." 1>&2
+        # Print the output in case the user redirect the log to console.
+        echo $output 1>&2
         exit 1
     fi
 
-    echo ${output}
+    if ! [[ $output =~ ^EXEC_PREFIX:.* ]]; then
+        echo "[ERROR] The output format is incorrect." 1>&2
```

Comment:
I would state clearly the last line of the output is excepted to be the BashJavaUtils execution result with the certain prefix. This will help people understand which contract is broken.